lib/internal/test_runner/test.js

```diff
@@ -122,6 +123,30 @@ class TestContext {
 
   constructor(test) {
     this.#test = test;
+
+    this.test = (name, options, fn) => {
+      const overrides = {
+        __proto__: null,
+        loc: getCallerLocation(),
+      };
+      // eslint-disable-next-line no-use-before-define
+      const subtest = this.#test.createSubtest(Test, name, options, fn, overrides);
+      return subtest.start();
+    };
```

Comment:
wouldn't it be more performant to leave this as a prototype method, and instead do a public class field `test = FunctionPrototypeBind(this.test, this);`, so that the "meat" of the method can be shared and optimized across all instances?

separately, couldn't the skip/todo/only methods be assigned to the prototype method once, in a static block, instead of a new function created on every instance?

Reply:
If you mean with Object.defineProperty, then without changing the `#test` to `[kTest]` it can't as it cannot access private method outside the class.
if you mean adding the skip/todo/any in the constructor but keeping the test function as is, than for some reason it did not work for me

The skip/todo/any methods are needed to assign on the test function, not on the `TestContext`

Reply:
Ah, true, the 3 expandos need to be done in the constructor every time - altho the meat could be shared private methods and the per-instance thing could just be a bound function.